### ISSUE TITLE
Image with S3 path not working

### ISSUE THREAD

**Issue by tatva-170** [Foliotek/Croppie#612]
## Expected Behavior
We are expecting that it should work with amazon S3 path.

## Actual Behavior
We are uploading images on amazon S3. So while editing an image we are using S3 path to display an image but we are getting CORS error in croppie. I already looked into other tickets related to this issue and tried all things but still its not working. Now the strange thing is, its not working for most of people but its working for my colleague for the same website and same environment.

Here is the screenshot of error we are facing:
![cors-error](https://user-images.githubusercontent.com/6491908/51375747-b32b9680-1b2c-11e9-9cc7-fa76b264baa9.jpg)

And screenshot of S3 CORS configuration
![s3-cors-configuration](https://user-images.githubusercontent.com/6491908/51375755-b9217780-1b2c-11e9-9141-a32b2a733bcb.jpg)

## Steps to Reproduce the Problem
We are able to reproduce an issue but as its after login so I can't share login credentials.

## Example Link
* Link: https://jsfiddle.net/1qgp4bjh/3
  
## Specifications
  - Browser: Chrome (Windows)
  - Version: 71.0.3578.98

**Comment by thedustinsmith** [comment]
Try opening up your network panel in chrome dev tools and inspecting the request to the image.  I'm curious what the response headers are for that image request.

**Comment by tatva-170** [comment]
Thanks for your reply Dustin.
Here is the screenshot of request.

![image-headers](https://user-images.githubusercontent.com/6491908/51426315-ed6d6480-1c0e-11e9-84c0-9dc64dfa3e00.jpg)
In Preview and Response tabs its blank.

**Comment by thedustinsmith** [comment]
Out of curiosity, can you try disabling your cache or try an incognito window?  I'm wondering if you requested the image before you had CORS enable on the bucket, now your browser isn't fetching it again.  Your settings all look correct to me, and the fiddle works fine.

**Comment by tatva-170** [comment]
Thanks for looking into it.
I have tried with cache clear and incognito window as well but still not working. And I have requested the image after applying CORS on the bucket.
What you think what the issue might be?

**Comment by chadhuber** [comment]
tldr, on your <img /> tag include the attribute crossorigin="anonymous" such as

`<img src="https://s3.amazonaws.com/bucket/img.jpg" crossorigin="anonymous" />`

**Explanation**
What's happening here is that the <img> tag loads the image without requesting any crosorigin data from S3, so it (incorrectly) doesn't provide it. Then, Croppie requests that img with crossorigin, but [Chrome ignores that request](https://stackoverflow.com/questions/49503171/the-image-tag-with-crossorigin-anonymous-cant-load-success-from-s3) and just uses the original (cached) result which doesn't contain any crossorigin data, so it prevents it. By putting the crossorigin attribute on the first load of the image, when Chrome uses the cached version, it will see the cross origin data that came from S3. One quick way to test all of this is to open your Chrome inspector and select 'Disable cache' under the 'Network' tab and it should work no problem.

**Comment by HoangTrung** [comment]
+1
